pkg/kn/commands/trigger/update.go

```diff
@@ -112,3 +110,16 @@ func NewTriggerUpdateCommand(p *commands.KnParams) *cobra.Command {
 
 	return cmd
 }
+
+func extractFilters(trigger *v1alpha1.Trigger) util.StringMap {
+	if trigger.Spec.Filter == nil {
+		return nil
```

Comment:
You should return an empty map here, otherwise you can a panic when you call `existing.Merge` and `existing` is nil. Its also inconsistent to return different things when `Filter == nil` or `Filter.Attributes == nil`

Reply:
Changed it in the latest version. Thank you.